js/index.js

```diff
@@ -70,24 +70,78 @@ function updateArticles() {
 
   function renderHTML(data) {
     var date = new Date();
+    var day = date.getDay();
     var hour = date.getHours();
-    var title, views;
-
-    var i = 1;
-    var counter = 0;
+    var minute = date.getMinutes();
+    switch("hour") {
+
+      // Uses the past day Json file and filters top articles from past 60 minutes.
+      case "hour":
+        var filtered = data.filter(function (a) {
+          var articleHour = a[2];
+          var articleMinute = a[3];
+          return (articleHour==hour-1 && articleMinute>=minute || articleHour==hour);
+        });
+        break;
+
+      // Uses the past 7 days Json file and filters top articles from past 24 hours.
+      case "day":
+
+        var filtered = data.filter(function (a) {
+          var articleHour = a[2];
+          var articleDay = a[3];
+          return (articleDay==day-1 && articleHour>=hour || articleDay==day);
+        });   
+        console.log(filtered);
+        break;
+        
+      // Displays past 7 day's top articles. Does not need to be filtered.
+      default:
+        var filtered = data;
+        
+    }
+    
+    // Adds the views together.
+    var toPrint = sumOfFiltered(filtered);
+    
+    toPrint.sort((function(a,b) {
+      return b[4] - a[4];
+    }));
+    update(toPrint);
+
+    
+    // Adds the views and returns to toPrint.
```

Comment:
Leaving comments to explain your code is a good practice, but this doesn't tell us anything that we couldn't have figured out from the name of the function. It would be more helpful if the comment briefly went over the logic at the top most level.